src/shims/unix/fd.rs

```diff
@@ -296,6 +296,12 @@ impl WeakFileDescriptionRef {
     }
 }
 
+impl VisitProvenance for WeakFileDescriptionRef {
+    fn visit_provenance(&self, _visit: &mut VisitWith<'_>) {
+        // All our WeakFileDescriptionRef do not have any tags.
+    }
+}
+
```

Comment:
I am not sure what is the purpose of ``VisitProvenance``. The compiler requires me to implement this in order to pass ``WeakFileDescriptionRef`` to ``callback!``.

Reply:
The purpose is to call the `visit` callbacks on all pointers and places stored anywhere in the machine state.

However, a *weak* reference can never be the only reference to some pointer or place, so it is okay to make this a NOP for weak references. On the full `FileDescriptionRef` type, this would have to visit the file description itself, which would require making `VisitProvenance` a supertrait of `FileDescription`... lucky enough it seems like we do not have to do that.

Please fix the comment according to what I wrote here.